Add kubestash docs (#1)
kubestash/docs: docs/guides/hooks/configuring-hooks/index.md

  containerName: mysql
```

## Based on Executor

Based on executor of a hook, we can configure them in the following ways.

### Pod Executor

Provide list of key value pair that will be used as label selector to select the desired pods. You can use comma to separate multiple labels (i.e. `"app=my-app,env=prod"`).

**Examples:**

- Execute hook on all the selected pods (only filtered by `selector`):

```yaml
executor:
  type: Pod
  pod:
    selector: env=dev, app=web
    strategy: ExecuteOnAll
```

- Execute hook on only one of the selected pods (only filtered by `selector`). This is default behavior:

```yaml
executor:
  type: Pod
  pod:
    selector: env=dev, app=web
```

- Execute hook on all the selected pods (filtered by `selector` and `owner`):

```yaml
executor:
  type: Pod
  pod:
    selector: env=dev, app=web
    owner:
      apiVersion: apps/v1
      kind: Deployment
      name: kubestash-op
    strategy: ExecuteOnAll
```

### Operator Executor

KubeStash operator itself will execute the hook. 

**Example:**

```yaml
executor:
  type: Operator
```

### Function Executor

> Hook is executed by a Job. We need to create a [Function](/docs/concepts/crds/function/index.md) define the `Function` and its parameters in [FunctionHookExecutorSpec](/docs/concepts/crds/hooktemplate/index.md#specexecutor) that will be used to create hook executor job.

Reviewer: ```suggestion
Hook is executed by a Job. We need to create a [Function](/docs/concepts/crds/function/index.md) and refer it in [FunctionHookExecutorSpec](/docs/concepts/crds/hooktemplate/index.md#specexecutor) that will be used to create hook executor job.
```
